### ISSUE TITLE
Reduce or remove generated JavaScript

### ISSUE THREAD

**Issue by hangy** [openfoodfacts/openfoodfacts-server#2672]
Right now, we have a lot of JavaScript that is generated in Perl in the form of `$ProductOpener::Display:scripts` and `$ProductOpener::Display:initjs`. Ideally, this could use unobtrusive JS approaches and avoid redundant JS (and sometimes CSS).

Potential benefits
- For front end designers, it will be more obvious where the UI behaviour is defined through JavaScript
- Less redundant JavaScript, ie. for `tagsInput` on `product.pl`
- Less chance of potentially breaking JavaScript code when changing Perl code
- Perl code should be easier to read
- Smaller HTML pages and potential to cache the JS/CSS files

Potential challenges
- We may need to export the (`.po`) translations from Perl to a JSON file in a somewhat efficient way (file per language?), so that they can be used for I18N in JavaScript

**Comment by Kirti2502** [comment]
Hello, I am an outreachy applicant and am well versed with javascript but learning perl. I would like to contribute to the project.
Is the the right issue to start my contribution? If not please guide me.
Thank you.

**Comment by stephanegigandet** [comment]
Hello @Kirti2502 !

To illustrate the problem, here is the code for form to edit a product on Open Food Facts:
https://github.com/openfoodfacts/openfoodfacts-server/blob/master/cgi/product_multilingual.pl

It's a Perl script that generate the HTML, + the associated Javascript and also some CSS styles.

It makes it quite difficult for people who don't know Perl to change the UI, improve the Javascript code etc.

But it's rather complex issue. I don't know really know myself what's the best way to solve it. Instead of generating specific javascript code (that varies for each product and dépends on the language selected by the user), we could perhaps try to create more generic javascript code that discovers what to do based on the data in the HTML.

Maybe @hangy has ideas on the best way to approach this. Ideally it would be good to do it step by step if possible.

**Comment by teolemon** [comment]
@stephanegigandet @hangy I believe this one is fixed, right ?

**Comment by stephanegigandet** [comment]
To some extent, we still have generated javascript.
